### PR TITLE
23697 - fix stage1 to stage2 dissolution transition bug

### PR DESCRIPTION
**Pull request by eason-pan-bc**
*Issue #:* /bcgov/entity#23697

*Description of changes:*
- Added the `disable-dissolution-sftp-bcmail` feature flag option in `flags.json`
- Updated the logic for furnishing entries check when moving from stage 1 to stage 2 of involuntary dissolution
    - check furnishing entries - 2 scenarios
        1. if email processed, and mail processed (after 5 business days still not in good standing)
        2. only has a mail entry, and it's processed
- Updated the check to support SFTP function FF

#### Local Test Results 
**Action**: Trying to move a business from stage 1 to stage 2 of involuntary dissolution

**Scenario 1** - ERROR and not moving to stage 2
_when this business has an email entry and haven't sent out a mail _
Pasted image 20241011153110
Pasted image 20241011153047

**Scenario 2** - SUCCESS moving to stage 2
_when this business has an email entry and after 5 business days still not in good standing, a mail sent out_
Pasted image 20241011153215
Pasted image 20241011153238

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: jobs/involuntary-dissolutions/involuntary_dissolutions.py [hunk 1]
```diff
@@ -239,16 +239,34 @@ def stage_2_process(app: Flask):
...
-            furnishing.furnishing_type in (Furnishing.FurnishingType.EMAIL, Furnishing.FurnishingType.MAIL)
+        email_processed = any(
+            furnishing.furnishing_type == Furnishing.FurnishingType.EMAIL
             and furnishing.status == Furnishing.FurnishingStatus.PROCESSED
             for furnishing in furnishings
         )
+        # if SFTP function is off, we expect the mail status will be QUEUED
+        expected_mail_status = Furnishing.FurnishingStatus.PROCESSED
+        if flags.is_on('disable-dissolution-sftp-bcmail'):
+            expected_mail_status = Furnishing.FurnishingStatus.QUEUED
```
Comment (chenhongjing) [inline jobs/involuntary-dissolutions/involuntary_dissolutions.py:258]: We will turn FF on and off for testing purpose, so you should make sure that both `PROCESSED` and `QUEUED` pass the check. For example, the mail furnishing is set processed in stage 1 with disable-FF off, and then we turn this flag on and move it into stage 2 and this is also expected to pass the check.

Comment (eason-pan-bc) [reply jobs/involuntary-dissolutions/involuntary_dissolutions.py:258]: good call! thanks for pointing it out 👍

File: jobs/furnishings/flags.json [hunk 2]
```diff
@@ -1,5 +1,6 @@
 {
     "flagValues": {
-        "enable-involuntary-dissolution": true
+        "enable-involuntary-dissolution": true,
+        "disable-dissolution-sftp-bcmail": true
```
Comment (JazzarKarim) [inline jobs/furnishings/flags.json:4]: Since you're here, I think you can also add the FF for BCLaws 🙏

**Review by JazzarKarim — COMMENTED** [review]
Eason, there's a small SonarCloud issue: https://sonarcloud.io/project/issues?id=bcgov_lear&pullRequest=3028&issueStatuses=OPEN,CONFIRMED&sinceLeakPeriod=true
image

If you can fix that, that would be awesome. You just might need to do a very small refactor for the function.

File: jobs/involuntary-dissolutions/involuntary_dissolutions.py (hunk 1 above)
Comment (eason-pan-bc) [reply jobs/involuntary-dissolutions/involuntary_dissolutions.py]: adjusted the logic, now it appends the QUEUED status for checking when FF is on

File: jobs/furnishings/flags.json (hunk 2 above)
Comment (eason-pan-bc) [reply jobs/furnishings/flags.json]: added 😄

**Comment by eason-pan-bc** [comment]
> Eason, there's a small SonarCloud issue: https://sonarcloud.io/project/issues?id=bcgov_lear&pullRequest=3028&issueStatuses=OPEN,CONFIRMED&sinceLeakPeriod=true image
> 
> If you can fix that, that would be awesome. You just might need to do a very small refactor for the function.

Refactored 😄 
image

File: jobs/involuntary-dissolutions/involuntary_dissolutions.py (hunk 1 above)
Comment (chenhongjing) [reply jobs/involuntary-dissolutions/involuntary_dissolutions.py]: Think you can also update tests a bit 👀

Comment (eason-pan-bc) [reply jobs/involuntary-dissolutions/involuntary_dissolutions.py]: haha, you read my mind, I'm adding tests, like now 😄

**Review by chenhongjing — APPROVED** [review]
LGTM

**Review by AimeeGao — APPROVED** [review]
LGTM(^_^)

**Pull request merged**